lib/Mojo/WebSocketProxy/Dispatcher.pm

```diff
@@ -66,7 +66,10 @@ sub open_connection {
         # We normalize the entire string before decoding.
         my $normalized_msg = Unicode::Normalize::NFC($msg);
         my $args = eval { decode_json_utf8($normalized_msg) };
-        $log->error("JSON decoding $normalized_msg failed: $@") if $@;
+        if ($@) {
```

Comment:
since this is a public module, I'd suggest:

```perl
if(my $args = eval { decode_json_utf8($normalized_msg) }) {
    on_message($c, $args);
} else {
    $log->debug("JSON decoding failed for: $normalized_msg");
    # maybe $c->emit(parse_error => $@, $normalized_msg);
    return;
};
```

Reply:
Thanks @binarytom !  There doesn't seem to be a `parse_error` event currently; I was thinking of emitting a regular [`error`](https://metacpan.org/pod/release/SRI/Mojolicious-7.29/lib/Mojo/EventEmitter.pm#error) instead, would that work?

Another way I'm currently handling this is just to guard with [`is_level`](https://metacpan.org/pod/release/SRI/Mojolicious-7.29/lib/Mojo/Log.pm#is_level):

```perl
if (my $args = ...) {
    ...
} else {
    $log->debug(...);
    $log->error($@) if $log->is_level('debug');
}
```

What do think?  cc @raunakkathuria

Reply:
OTOH, the `else` side of the above won't work (even with the `return;` as we do not emit a `BadRequest` response (which comes from `Dispatcher::on_message()/Parser::parse_req()`.

Reply:
I think better to have consistent logging - if it only appears at `debug` level, it should be a `debug` message.

If we receive a request which fails at the decoding step, I'd suggest just dropping the connection - we can send back a generic error first, "malformed JSON" for example.

Reply:
Thanks, I've opted to follow the [spec](https://tools.ietf.org/html/rfc6455#section-7.1.7) on closing the connection here and logging to debug.